lib/cli/src/commands/compile.rs

```diff
@@ -105,15 +106,26 @@ impl Compile {
 
         #[cfg(feature = "object-file")]
         if engine_type == EngineType::ObjectFile {
-            let artifact: &wasmer_engine_object_file::ObjectFileArtifact =
-                module.artifact().as_ref().downcast_ref().context("Engine type is ObjectFile but could not downcast artifact into ObjectFileArtifact")?;
-            let symbol_registry = artifact.symbol_registry();
-            let metadata_length = artifact.metadata_length();
+            use wasmer_engine_object_file::ObjectFileArtifact;
+
+            let symbol_registry = unsafe {
+                module.query_artifact(|artifact: &ObjectFileArtifact| {
+                    Some(Arc::clone(artifact.symbol_registry()))
+                })
+            };
+            let metadata_length = unsafe {
+                module.query_artifact(|artifact: &ObjectFileArtifact| {
+                    Some(Arc::new(artifact.metadata_length()))
```

Comment:
For example, the use of `Arc` here seems unnecessary

Reply:
We may also be able to do this with 1 unsafe block, returning a tuple of both things we care about

Reply:
Yup, using `Arc` here is… stupid :-/.